docs/source/docs/configuration.md

```diff
@@ -2,19 +2,592 @@
+your own networks to pass to DeepReg train via config.
+
+The `num_channel_initial` is used to define the number of initial channels for the
+network, and should be int type.
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "unet" # One of unet, local, global: networks currently supported by DeepReg
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+```
+
+#### UNet
+
+The UNet model requires several additional arguments to define it's structure:
+
+- `depth`: int, defines the depth of the UNet from first to bottom, bottleneck layer.
+- `pooling`: Boolean, pooling method used for downsampling. True: non-parametrized
+  pooling will be used, False: conv3d will be used.
+- `concat_skip`: Boolean, concatenation method for skip layers in UNet. True:
+  concatenation of layers, False: addition is used instead.
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "unet" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    depth: 3
+    pooling: false
+    concat_skip: true
+```
+
+#### Local and GlobalNet
+
+The LocalNet has an encoder-decoder structure, and extracts information from tensors at
+certain levels. We can define which levels to extract info from with the
+`extract_levels` argument.
+
+The GlobalNet encodes the image and uses the bottleneck layer to output an affine
+transformation using a FCN.
+
+- `extract_levels`: list of positive ints (ie, the min value in `extract_levels` should
+  be >=0). Eg. [1, 2, 3, 5] will extract information at those levels (but not 4).
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "local" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    extract_levels: [0, 1, 2]
+```
+
+### Loss - required
+
+This section defines the loss in training.
+
+The losses in DeepReg are defined depending on the type of network to be built, and can
+be split into three sections: imag and label losses (between moving and fixed tensors),
+and regularization losses (on the DDFs predicted).
+
+DeepReg uses `tf.keras.Model` `add_loss()` in the Registry method to add losses to the
+model, which provides some flexibility in configuration.
+
+Currently, DeepReg offers conditional, ddf/dvf and affine registration pre-built models.
+Traditionally, models have been configured with the following losses:
+
+- Conditional: label loss.
+- DDF/DVF: ddf loss, image loss, label loss.
+- Affine: ddf loss, image loss, label loss.
+
+The above sections are necessary to build a model correctly. Not passing all sections as
+defined above may raise errors. Currently you can call one loss per field eg. one label
+loss type, one image loss type and one ddf/dvf loss type.
```

Comment:
also: you can set weight to 0 to not use it. weights on label loss will be ignored if the "labeled" key was false or segmentation labels are not available.